var/spack/repos/builtin/packages/vtkm/package.py

```diff
@@ -17,39 +17,77 @@ class Vtkm(Package):
     architectures."""
 
     homepage = "https://m.vtk.org/"
-    url      = "https://gitlab.kitware.com/api/v4/projects/vtk%2Fvtk-m/repository/archive.tar.gz?sha=v1.1.0"
+    url      = "https://gitlab.kitware.com/api/v4/projects/vtk%2Fvtk-m/repository/archive.tar.gz?sha=v1.3.0"
     git      = "https://gitlab.kitware.com/vtk/vtk-m.git"
 
     version('master', branch='master')
+    version('1.3.0', "d9f6e274dec2ea01273cccaba356d23ca88c5a25")
+    version('1.2.0', "3295fed86012226c107e1f2605ca7cc583586b63")
     version('1.1.0', "6aab1c0885f6ffaaffcf07930873d0df")
 
     variant("cuda", default=False, description="build cuda support")
+    variant("logging", default=False, description="build logging support")
+    variant("mpi", default=False, description="build mpi support")
+    variant("openmp", default=False, description="build openmp support")
+    variant("rendering", default=True, description="build rendering support")
     variant("tbb", default=True, description="build TBB support")
 
     depends_on("cmake")
     depends_on("tbb", when="+tbb")
     depends_on("cuda", when="+cuda")
+    depends_on("mpi", when="+mpi")
 
     def install(self, spec, prefix):
         with working_dir('spack-build', create=True):
             cmake_args = ["../",
                           "-DVTKm_ENABLE_TESTING=OFF",
-                          "-DVTKm_BUILD_RENDERING=ON",
                           "-DVTKm_USE_64BIT_IDS=OFF",
                           "-DVTKm_USE_DOUBLE_PRECISION=ON"]
-            # tbb support
-            if "+tbb" in spec:
-                # vtk-m detectes tbb via TBB_ROOT env var
-                os.environ["TBB_ROOT"] = spec["tbb"].prefix
-                cmake_args.append("-DVTKm_ENABLE_TBB=ON")
 
             # cuda support
             if "+cuda" in spec:
                 cmake_args.append("-DVTKm_ENABLE_CUDA=ON")
-                # this fix is necessary if compiling platform has cuda, but
-                # no devices (this common for front end nodes on hpc clusters)
-                # we choose kepler as a lowest common denominator
-                cmake_args.append("-DVTKm_CUDA_Architecture=kepler")
+                if 'cuda_arch' in spec.variants:
```

Comment:
I don't see where this `cuda_arch` variant is defined.

Reply:
It serves as an additional command line argument. I just follow the pattern in spack(Ex. accfft and dealii). Ideas?

Reply:
Both of those packages subclass `CudaPackage` in addition to the regular base class. If you add this, then `cuda_arch` will be defined.